jardin/instrumentation/notifier.py

```diff
@@ -0,0 +1,45 @@
+import uuid
+from threading import Lock
+from jardin import config as config
+from jardin.instrumentation.base_subscriber import BaseSubscriber
+from jardin.instrumentation.event import Event
+
+class NullNotifier:
```

Comment:
Proposal for further simplification:

1. delete the `JARDIN_INSTRUMENTATION_ENABLED` env var
2. delete this `NullNotifier` class
3. rename `SimpleNotifier` to `Notifier`
4. In `Notifier.report_event`, without taking the lock, return early if there are zero subscribers

And maybe add a comment to say that we are intentionally not taking the lock because the race condition isn't worth it.